heron/executor/tests/python/heron_executor_unittest.py

```diff
@@ -44,31 +48,63 @@ def _run_process(self, name, cmd, env=None):
+           "-XX:TargetSurvivorRatio=90 -XX:+PrintCommandLineFlags -verbosegc -XX:+PrintGCDetails " \
+           "-XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintGCCause " \
+           "-XX:+PrintPromotionFailure -XX:+PrintTenuringDistribution -XX:+PrintHeapAtGC " \
+           "-XX:+HeapDumpOnOutOfMemoryError -XX:+UseConcMarkSweepGC -XX:ParallelGCThreads=4 " \
+           "-Xloggc:log-files/gc.%s.log -XX:+HeapDumpOnOutOfMemoryError " \
+           "-Djava.net.preferIPv4Stack=true -cp instance_classpath:classpath " \
+           "com.twitter.heron.instance.HeronInstance topname topid %s %s %d 0 stmgr-%d " \
+           "master_port metricsmgr_port heron/config/src/yaml/conf/test/test_heron_internals.yaml" \
+           % (instance_name, instance_name, component_name, instance_id, container_id)
 
   MockPOpen.set_next_pid(37)
   expected_processes_container_0 = [
-    ProcessInfo(MockPOpen(), 'heron-shell-0', shell_command_expected),
-    ProcessInfo(MockPOpen(), 'metricsmgr-0',  get_expected_metricsmgr_command(0)),
-    ProcessInfo(MockPOpen(), 'heron-tmaster', 'tmaster_binary master_port tmaster_controller_port tmaster_stats_port topname topid zknode zkroot stmgr-1 heron/config/src/yaml/conf/test/test_heron_internals.yaml metrics_sinks_config_file metricsmgr_port'),
+      ProcessInfo(MockPOpen(), 'heron-shell-0', shell_command_expected),
+      ProcessInfo(MockPOpen(), 'metricsmgr-0', get_expected_metricsmgr_command(0)),
+      ProcessInfo(MockPOpen(), 'heron-tmaster',
+                  'tmaster_binary master_port '
+                  'tmaster_controller_port tmaster_stats_port '
+                  'topname topid zknode zkroot stmgr-1 '
+                  'heron/config/src/yaml/conf/test/test_heron_internals.yaml '
+                  'metrics_sinks_config_file metricsmgr_port'),
   ]
 
   MockPOpen.set_next_pid(37)
   expected_processes_container_1 = [
-    ProcessInfo(MockPOpen(), 'stmgr-1', 'stmgr_binary topname topid topdefnfile zknode zkroot stmgr-1 container_1_word_3,container_1_exclaim1_2,container_1_exclaim1_1 master_port metricsmgr_port shell-port heron/config/src/yaml/conf/test/test_heron_internals.yaml'),
-    ProcessInfo(MockPOpen(), 'container_1_word_3', get_expected_instance_command('word', 3)),
-    ProcessInfo(MockPOpen(), 'container_1_exclaim1_1', get_expected_instance_command('exclaim1', 1)),
-    ProcessInfo(MockPOpen(), 'container_1_exclaim1_2', get_expected_instance_command('exclaim1', 2)),
-    ProcessInfo(MockPOpen(), 'heron-shell-1', shell_command_expected),
-    ProcessInfo(MockPOpen(), 'metricsmgr-1', get_expected_metricsmgr_command(1)),
+      ProcessInfo(MockPOpen(), 'stmgr-1',
+                  'stmgr_binary topname topid topdefnfile zknode zkroot stmgr-1 '
+                  'container_1_word_3,container_1_exclaim1_2,container_1_exclaim1_1 master_port '
+                  'metricsmgr_port shell-port '
+                  'heron/config/src/yaml/conf/test/test_heron_internals.yaml'),
```

Comment:
Agree. Maybe you can refer to [how I did this in explorer](https://github.com/twitter/heron/blob/master/heron/explorer/tests/python/explorer_unittest.py#L30)?